client/src/components/modules/Navbar/SettingsDialog.tsx

```diff
@@ -12,18 +12,21 @@ export type SettingsDialogProps = {
   setVisible: (boolean) => void;
 };
 
-let suppressedTimezoneUpdate = localStorage.getItem("suppressedTimezoneUpdate");
 function TimezoneModal(props) {
   const { onOk, onCancel, visible, user, timezone, loading } = props;
   function displayTimezoneOffset(offset) {
     return offset < 0 ? `UTC${offset}` : `UTC+${offset}`;
   }
+  let suppressedTimezoneUpdate = localStorage.getItem("suppressedTimezoneUpdate");
 
   return (
     <Modal
       okText="Yes"
       cancelText="No"
-      onOk={onOk}
+      onOk={() => {
+        localStorage.setItem("suppressedTimezoneUpdate", timezone);
```

Comment:
do you think this is necessary? seems more correct for this to only be set in `onCancel`

Reply:
also only line 59, i feel like that shouldn't be state. if that's behaving weird, maybe it would be better if we change it to:
```
const timezoneDiffers = props.user.timezone !== undefined && browserTimezone != props.user.timezone
```

Reply:
it's prooobably not necessary, but I thought it would be weird when the dialog pops up again if the user changes their timezone setting to something else (for whatever reason) after clicking OK.

sure, I will change the prop to a local variable